### PR TITLE
Added CONTRIBUTING

### PR DESCRIPTION
**Pull request by lunny**
For building a healthy and fast development community, I drafted this CONTRIBUTING and please send PR to improve it. According to the CONTRIBUTING, CONTRIBUTING's change should all of gitea owners' approval.

### CODE REVIEW

File: CONTRIBUTORS
Comment (bkcsoft) [inline CONTRIBUTORS]: `who has contributed`?

File: MAINTAINERS
Comment (bkcsoft) [inline MAINTAINERS]: `who can review pull-requests` ?

**Review by bkcsoft — COMMENTED** [review]
To avoid confusion, these should be different.

**Comment by bkcsoft** [comment]
LGTM

File: CONTRIBUTING.md
Comment (strk) [inline CONTRIBUTING.md]: compatibility when upgrade.

Comment (strk) [reply CONTRIBUTING.md]: Or better: compatibility on upgrade.

File: CONTRIBUTING.md
Comment (strk) [inline CONTRIBUTING.md]: . . an owner or a maintainer.

File: CONTRIBUTING.md
Comment (strk) [inline CONTRIBUTING.md]: ... to ensure every PR is reviewed by at least 2 maintainers. 

(Abd turn lgtm to a link maybe)

File: CONTRIBUTING.md
Comment (strk) [inline CONTRIBUTING.md]: Contributors

File: CONTRIBUTING.md
Comment (strk) [inline CONTRIBUTING.md]: ^more than 4 PRs(included)^at least 4 PR

File: CONTRIBUTING.md
Comment (strk) [inline CONTRIBUTING.md]: As two years are mentioned here, it could make sense to also set a date for renewal. November 1st ?

File: CONTRIBUTING.md
Comment (strk) [inline CONTRIBUTING.md]: I actually think it is needed to add an year of last change as in Copyright (C) 2014-2016. And use multiple lines for multiple copyright owners (Gogs and Gitea contributors..)

File: MAINTAINERS
Comment (strk) [inline MAINTAINERS:1]: Who can review and approve changes

File: CONTRIBUTING.md
Comment (tboerger) [inline CONTRIBUTING.md]: change or add something to the

File: CONTRIBUTING.md
Comment (tboerger) [inline CONTRIBUTING.md]: Before starting to write...

File: CONTRIBUTING.md
Comment (tboerger) [inline CONTRIBUTING.md]: sent a PR to Gitea that gets accepted will be a contributor.

Comment (tboerger) [reply CONTRIBUTING.md]: Please send a PR to add your name to

File: CONTRIBUTING.md
Comment (tboerger) [inline CONTRIBUTING.md]: I would skip the pr numbers part.

File: CONTRIBUTING.md
Comment (tboerger) [inline CONTRIBUTING.md]: To make sure every PR have been checked...

Comment (tboerger) [reply CONTRIBUTING.md]: Any PR must be reviewed by at least two maintainers before it can get merged.

File: CONTRIBUTING.md
Comment (tboerger) [reply CONTRIBUTING.md]: Gitea

Comment (tboerger) [reply CONTRIBUTING.md]: ...at least 4 accepted PRs.

Comment (tboerger) [reply CONTRIBUTING.md]: contributor

File: CONTRIBUTING.md
Comment (tboerger) [inline CONTRIBUTING.md]: And the owners or the team maintainer...

Comment (tboerger) [reply CONTRIBUTING.md]: Should spend some time on code reviews.

File: CONTRIBUTING.md
Comment (tboerger) [inline CONTRIBUTING.md]: I'm glad to be an owner..,

File: CONTRIBUTING.md
Comment (tboerger) [inline CONTRIBUTING.md]: Gitea

Comment (tboerger) [reply CONTRIBUTING.md]: without any company support

Comment (tboerger) [reply CONTRIBUTING.md]: to keep the development

File: CONTRIBUTING.md
Comment (tboerger) [inline CONTRIBUTING.md]: If some owner don't obey these rules, the other owners are allowed to revoke his owner status.

File: CONTRIBUTING.md
Comment (tboerger) [inline CONTRIBUTING.md]: he agrees with these rules on the

**Review by tboerger — CHANGES_REQUESTED** [review]

**Review by tboerger — CHANGES_REQUESTED** [review]
The text within contributors and maintainers files are wrong, we don't have a cla. Beside that I would also suggest to add the rule that we add the signed off by line to every commit, that's better than a cla for oss projects.

**Comment by lunny** [comment]
@tboerger done

**Comment by tboerger** [comment]
I think we should add also https://github.com/kleister/kleister-api/blob/master/DCO and write something like this to the contributors guide:

```

## Sign your work

The sign-off is a simple line at the end of the explanation for the patch. Your
signature certifies that you wrote the patch or otherwise have the right to pass
it on as an open-source patch. The rules are pretty simple: If you can certify
[DCO](DCO), then you just add a line to every git commit message:

```

Signed-off-by: Joe Smith joe.smith@email.com

```

Please use your real name, we really dislike pseudonyms or anonymous
contributions. We are in the opensource world without secrets. If you set your
`user.name` and `user.email` git configs, you can sign your commit automatically
with `git commit -s`.
```

You can see an example at https://github.com/kleister/kleister-api/blob/master/CONTRIBUTING.md

File: MAINTAINERS
Comment (tboerger) [reply MAINTAINERS:1]: I would drop all the comments.

File: CONTRIBUTORS
Comment (tboerger) [inline CONTRIBUTORS:1]: I would drop all the comments.

File: CONTRIBUTING.md
Comment (tboerger) [inline CONTRIBUTING.md]: Please add a space behind the name, maybe we should link to the GitHub profiles instead of linking to email addresses and use the real names :)

**Review by tboerger — CHANGES_REQUESTED** [review]

File: CONTRIBUTORS
Comment (lunny) [reply CONTRIBUTORS:1]: Why remove all the comments? The information about format will be helpful?

File: CONTRIBUTING.md
Comment (strk) [reply CONTRIBUTING.md]: I hope one day we can get away from github, so wouldn't add more dependencies on it

File: CONTRIBUTORS
Comment (tboerger) [reply CONTRIBUTORS:1]: IMHO the format is obvious :)

**Review by tboerger — APPROVED** [review]

**Comment by tboerger** [comment]
LGTM

**Pull request merged**
